tests/test_data_index.py

```diff
@@ -111,6 +111,13 @@ def test_data_index_deps(self):
                 assert len(profiles) == len(deps["profiles"]), err_msg
                 assert len(params) == len(deps["params"]), err_msg
                 # assert correct dependencies are queried
-                assert queried_deps[p][name]["data"] == data | axis_limit_data, err_msg
+                # TODO: conversion from rpz to xyz is taken out of actual function
+                #       registration because of this data["phi"] is not queried in
+                #       the source code but actually needed for the computation. This
+                #       is a temporary fix until we have a better way to automatically
+                #       handle this.
+                assert queried_deps[p][name]["data"].issubset(
+                    data | axis_limit_data
+                ), err_msg
```

Comment:
If you can't find a better way, then let's instead subtract phi from the later set. FYI we removed the tests that ensures everything can be computed individually, i.e. without computing the entire dependency tree, and replaced it with this test which explicitly checks for this via inspecting the source code. For functions like the third derivatives of the basis vectors it's useful to keep this check as strict as possible.

Reply:
Substracting "phi" may cause problems since we only want to remove it if the quantity is one of the 3d vectors which changed basis in the main compute function. 
This was a temporary change. i will think of a better way to do it, once I finish the actual coding part.

Reply:
I think this should be ok? We're ensuring that the dependencies are a subset of what actually gets computed right? as opposed to ensuring equality?

Reply:
Actually since we deleted `phi` from dependencies, the previous version of this test might still be passing.

Reply:
@YigitElma In that case can you change this back

Reply:
Nevermind I changed it back on my branch.